### PR TITLE
117064357 Updating and deleting address 

### PR DESCRIPTION
**Pull request by sofiengwin**
#### What does this PR do?

It allows users to update or delete their address.
#### Description of Task to be completed?

Add address controller with actions for updating and destroying addresses.
#### How should this be manually tested?

During the checkout process, a user can decide to update or delete their address. 
#### Any background context you want to provide?

The destroy action uses a soft delete method where addresses are archived instead been destroy.
#### What are the relevant pivotal tracker stories?

```
#117064357
```

### CODE REVIEW

File: app/controllers/addresses_controller.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    @address = Address.find(params[:id])
+    respond_to do |format|
+      format.js { render "edit_address_form" }
+    end
+  end
+
+  def update
+    address = Address.find(params[:id])
+    if address.update_attributes(address_params)
+      redirect_to "/orders/summary/#{address.id}"
```
Comment (lekansogunle) [inline app/controllers/addresses_controller.rb]: Can we use use an alias for this. Refer to http://guides.rubyonrails.org/routing.html#overriding-the-named-helpers.

File: app/views/orders/_old_addresses.html.erb [hunk 2]
```diff
@@ -6,9 +6,16 @@
   <%= form_tag({controller: "orders", action: "summary"}, method: "post") do %>
   <div class="col l4">
     <div class="user-address">
-
       <div class="address-name">
         <%= address.name %>
+        <div class="manage-address">
+          <%= link_to "/addresses/#{address.id}", method: :delete, class: "delete-address-icon", data: { confirm: "Are you sure?" } do %>
```
Comment (lekansogunle) [inline app/views/orders/_old_addresses.html.erb]: Consider using a named helper path.

File: app/views/orders/_old_addresses.html.erb [hunk 3]
```diff
@@ -6,9 +6,16 @@
...
+          <i class="material-icons">delete</i>
+          <% end %>
+          <%= link_to "/addresses/#{address.id}/edit", remote: true, class: "edit-address-icon" do %>
```
Comment (lekansogunle) [inline app/views/orders/_old_addresses.html.erb]: Consider using a named helper path.

File: spec/controllers/addresses_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+RSpec.describe AddressesController, type: :request do
+  before(:all) do
+    @address = create(:address)
+  end
+
+  describe "POST update" do
+    before do
+      patch "/addresses/#{@address.id}", address: {
+        name: "James Longo",
```
Comment (lekansogunle) [inline spec/controllers/addresses_controller_spec.rb]: Try using the Faker gem.

File: spec/support/address_helpers.rb [hunk 5]
```diff
@@ -14,7 +14,7 @@ def fill_in_location_info(address)
 
   def fill_in_personal_info(address)
     fill_in "name", with: address.name
```
Comment (lekansogunle) [inline spec/support/address_helpers.rb]: For consistency, consider starting name with capital letter too.

File: app/controllers/addresses_controller.rb (hunk 1 above)
Comment (sofiengwin) [inline app/controllers/addresses_controller.rb]: Fixed. Using named helpers as recommended.

File: app/views/orders/_old_addresses.html.erb (hunk 2 above)
Comment (sofiengwin) [inline app/views/orders/_old_addresses.html.erb]: Fixed

File: app/views/orders/_old_addresses.html.erb (hunk 3 above)
Comment (sofiengwin) [inline app/views/orders/_old_addresses.html.erb]: Fixed

File: spec/controllers/addresses_controller_spec.rb (hunk 4 above)
Comment (sofiengwin) [inline spec/controllers/addresses_controller_spec.rb]: Fixed

File: app/models/address.rb [hunk 6]
```diff
@@ -6,4 +6,9 @@ class Address < ActiveRecord::Base
   validates :email, format: {
     with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\Z/
   }
+
+  def archive
+    self.archived_at = Time.now
+    save
```
Comment (lekansogunle) [inline app/models/address.rb:12]: NICE!

File: app/controllers/addresses_controller.rb [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    @address = Address.find(params[:id])
+    respond_to do |format|
+      format.js { render "edit_address_form" }
+    end
+  end
+
+  def update
+    address = Address.find(params[:id])
+    if address.update_attributes(address_params)
+      redirect_to address_summary_orders_path(address.id)
```
Comment (lekansogunle) [inline app/controllers/addresses_controller.rb]: Exception Handling! What if update attribute fails what happens. You might want to redirect to the edit page?

File: app/views/orders/_old_addresses.html.erb [hunk 8]
```diff
@@ -6,9 +6,16 @@
   <%= form_tag({controller: "orders", action: "summary"}, method: "post") do %>
   <div class="col l4">
     <div class="user-address">
-
       <div class="address-name">
         <%= address.name %>
+        <div class="manage-address">
+          <%= link_to address_path(address.id), method: :delete, class: "delete-address-icon", data: { confirm: "Are you sure?" } do %>
```
Comment (lekansogunle) [inline app/views/orders/_old_addresses.html.erb]: Good use of helper path.

**Comment by lekansogunle** [comment]
Except for the exception handling on app/controllers/addresses_controller.rb. LGTM!

**Comment by makinwab** [comment]
This PR has been reviewed but based on the new requirements and stories from the PO is of low priority.

**Pull request closed without merging**